Enhance aiRunnerImage flag

RUFFY-369 opened the pull request:
**What does this pull request do? Explain your changes. (required)**
Towards closing ['livepeer/bounties/#69](https://github.com/livepeer/bounties/issues/69)

This PR improves the `aiRunnerImage` flag to accept both string and JSON string inputs by adding the doc. Functionality is on the `ai-worker` side.

Usage: `aiRunnerImage="{'segment-anything-2': 'livepeer/ai-runner:segment-anything-2-v1.0'}"`

**Specific updates (required)**
- Usage doc has been added for the flag

**How did you test each of these updates (required)**

Test was performed by starting go-livepeer as aiWorker with `aiRunnerImage="{'segment-anything-2': 'livepeer/ai-runner:segment-anything-2-v1.0'}"` and verified launch version.


**Does this pull request close any open issues?**
<!-- Fixes # -->


**Checklist:**
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] Read the [contribution guide](./CONTRIBUTING.md)
- [x] `make` runs successfully
- [ ] All tests in `./test.sh` pass
- [x] README and other documentation updated
- [ ] [Pending changelog](./CHANGELOG_PENDING.md) updated

cc @rickstaa

RUFFY-369 committed:
chore:update help information regarding aiRunnerImage flag for also supporting JSON string

rickstaa commented on cmd/livepeer/livepeer.go:
```diff
@@ -160,7 +160,7 @@ func parseLivepeerConfig() starter.LivepeerConfig {
 	cfg.AIModels = flag.String("aiModels", *cfg.AIModels, "Set models (pipeline:model_id) for AI worker to load upon initialization")
 	cfg.AIModelsDir = flag.String("aiModelsDir", *cfg.AIModelsDir, "Set directory where AI model weights are stored")
-	cfg.AIRunnerImage = flag.String("aiRunnerImage", *cfg.AIRunnerImage, "Set the docker image for the AI runner: Example - livepeer/ai-runner:0.0.1")
+	cfg.AIRunnerImage = flag.String("aiRunnerImage", *cfg.AIRunnerImage, "Set the docker image for the AI runner: Example - livepeer/ai-runner:0.0.1 . You can also provide a JSON string to map specific pipeline names to custom images, allowing for pipeline-specific image overrides: Example - '{\"segment-anything-2\": \"livepeer/ai-runner:segment-anything-2-v1.0\", \"another-pipeline\": \"livepeer/ai ...
```
```suggestion
	cfg.AIRunnerImage = flag.String("aiRunnerImage", *cfg.AIRunnerImage, `Set the docker image for the AI runner: Example - livepeer/ai-runner:0.0.1 . You can also provide a JSON string to map specific pipeline names to custom images, allowing for pipeline-specific image overrides: Example - '{"segment-anything-2": "livepeer/ai-runner:segment-anything-2-v1.0", "another-pipeline": "livepeer/ai-runner:another-pipeline-v2.0"}'`)
```

rickstaa replied:
@RUFFY-369 looks good! However, maybe we can use raw string literals to cleanup the code a bit?

RUFFY-369 replied:
@rickstaa Just pushed the commit for the suggestion. Thanks!

rickstaa replied:
Thanks approved!

RUFFY-369 committed:
chore:add suggested formatting changes

rickstaa reviewed (approved):
LGTM 🙏🏻.

ad-astra-video commented:
@rickstaa @RUFFY-369 Are we waiting on anything to merge this PR?

RUFFY-369 commented:
> @rickstaa @RUFFY-369 Are we waiting on anything to merge this PR?

@ad-astra-video In my opinion, I don't think so. We can have @rickstaa 's thought if he has something in mind before this merge

rickstaa commented on cmd/livepeer/livepeer.go:
```diff
@@ -161,7 +161,7 @@ func parseLivepeerConfig() starter.LivepeerConfig {
 	cfg.AIModels = flag.String("aiModels", *cfg.AIModels, "Set models (pipeline:model_id) for AI worker to load upon initialization")
 	cfg.AIModelsDir = flag.String("aiModelsDir", *cfg.AIModelsDir, "Set directory where AI model weights are stored")
-	cfg.AIRunnerImage = flag.String("aiRunnerImage", *cfg.AIRunnerImage, "Set the docker image for the AI runner: Example - livepeer/ai-runner:0.0.1")
+	cfg.AIRunnerImage = flag.String("aiRunnerImage", *cfg.AIRunnerImage, `Set the docker image for the AI runner: Example - livepeer/ai-runner:0.0.1 . You can also provide a JSON string to map specific pipeline names to custom images, allowing for pipeline-specific image overrides: Example - '{"segment-anything-2": "livepeer/ai-runner:segment-anything-2-v1.0", "another-pipeline": "livepeer/ai-runner ...
```
@victorges do we want to rename this to `AIRunnerImageOverrides` and set a deprecation warning on this flag or are we good with keeping the current naming?

rickstaa committed:
feat: deprecate `aiRunnerImage` flag in favor of `aiRunnerImageOverrides`

This commit introduces deprecation logic for the `aiRunnerImage` flag, replacing
it with a new `aiRunnerImageOverrides` flag. The new flag is designed to support
enhanced image override functionality as implemented in the worker logic
in [ai-worker PR #293](https://github.com/livepeer/ai-worker/pull/293).

victorges commented on cmd/livepeer/starter/starter.go:
```diff
@@ -1211,7 +1214,22 @@ func StartLivepeer(ctx context.Context, cfg LivepeerConfig) {
-		n.AIWorker, err = worker.NewWorker(*cfg.AIRunnerImage, gpus, modelsDir)
+		// Retrieve image overrides from the config.
+		var imageOverrides worker.ImageOverrides
+		if err := json.Unmarshal([]byte(*cfg.AIRunnerImageOverrides), &imageOverrides); err != nil {
```
Probably should only do this if `*cfg.AIRunnerImageOverrides != ""`

rickstaa replied:
Good catch. I still have to run an end-to-end test but this will indeed fail. Will handle that logic 👍🏻.

victorges reviewed (approved):
LGTM just a small fix otherwise i think startup will fail without the overrides flag

rickstaa committed:
fix: ensure imageOverrides passes when empty

This commit ensures that the software functions correctly when imageOverrides
is empty. It also updates the description of the `aiRunnerImageOverrides` command
line flag.

rickstaa committed:
chore(ai): update ai-worker version

This commit ensures that the right ai-worker version is linked.
